signinapp/model.py

```diff
@@ -291,6 +303,33 @@ def get_from(cls, name: str, phone_number: str, email: str, contact_order: int)
         return guardian_user_data
 
 
+class Student(db.Model):
+    __tablename__ = "students"
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey("users.id"))
+
+    # Extra student information
+    graduation_year = db.Column(db.Integer)
+
+    # One to One: Links User to extra student information
+    user = db.relationship("User", back_populates="student_user_data")
+
+    @property
+    def display_grade(self):
+        grades = generate_grade_choices()
+        return grades[self.graduation_year]
+
+    @classmethod
+    def make(cls, name: str, password: str, graduation_year: int, **kwargs) -> User:
+        role = Role.from_name("student")
+        student = User.make(name=name, password=password, role=role, **kwargs)
+        student_user_data = Student(user_id=student.id, graduation_year=graduation_year)
```

Comment:
Is `student.id` populated at this point? I thought it only happened on a commit?

Reply:
Yeah I do a sneaky "db.session.flush()" in `User.make` so that the ID exists.